gpMgmt/bin/gprebalance_modules/planner.py

```diff
@@ -1,19 +1,476 @@
+        """
+        Pretty print shrink plan with segments to remove and moves to execute
+        """
+        lines = []
+        lines.append("=" * 80)
+        lines.append("SHRINK PLAN".center(80))
+        lines.append("=" * 80)
+        
+        # Target segment count
+        lines.append(f"\nTarget Segment Count: {self.target_segment_count}")
+        
+        # Shrinked segments section
+        lines.append(f"\n{'SEGMENTS TO REMOVE':-^80}")
+        if self.shrinked_segments:
+            lines.append(f"Total segments to shrink: {len(self.shrinked_segments)}")
+            lines.append("")
+            
+            for idx, seg_pair in enumerate(self.shrinked_segments, 1):
+                lines.append(f"  [{idx}] Segment Pair:")
+                
+                # Primary segment
+                if seg_pair.primaryDB:
+                    lines.append(f"      Primary:")
+                    lines.append(f"        Content:  {seg_pair.primaryDB.content}")
+                    lines.append(f"        DbId:     {seg_pair.primaryDB.dbid}")
+                    lines.append(f"        Host:     {seg_pair.primaryDB.hostname}")
+                    lines.append(f"        Datadir:  {seg_pair.primaryDB.datadir}")
+                    lines.append(f"        Port:     {seg_pair.primaryDB.port}")
+                
+                # Mirror segment
+                if seg_pair.primaryDB:
+                    lines.append(f"      Mirror:")
+                    lines.append(f"        Content:  {seg_pair.primaryDB.content}")
+                    lines.append(f"        DbId:     {seg_pair.primaryDB.dbid}")
+                    lines.append(f"        Host:     {seg_pair.primaryDB.hostname}")
+                    lines.append(f"        Datadir:  {seg_pair.primaryDB.datadir}")
+                    lines.append(f"        Port:     {seg_pair.primaryDB.port}")
+                
+                if idx < len(self.shrinked_segments):
+                    lines.append("")
+        else:
+            lines.append("  No segments to remove.")
+        
+        # Moves section
+        lines.append(f"\n{'BALANCE MOVES':-^80}")
+        if self.moves:
+            lines.append(f"Total moves planned: {len(self.moves)}")
+            lines.append("")
+            
+            for idx, move in enumerate(self.moves, 1):
+                lines.append(f"  [{idx}] {move}")
+                if idx < len(self.moves):
+                    lines.append("")
+        else:
+            lines.append("  No data migration needed.")
+        
+        lines.append("\n" + "=" * 80)
+        
+        return "\n".join(lines)
+    
+class ConfigurationEncoder:
+
+    @staticmethod
+    def encode_configutaion(gparray: gparray.GpArray, 
+                            target_hosts: List[Host],
+                            strategy: str) -> Tuple[Encoding, HostMapping]:
+        """
+        The rebalance solvers work with abstract input segment configutation.

```

Comment:
```suggestion
        The rebalance solvers work with abstract input segment configuration.
```
and please check other comments for typos.